### PR TITLE
NTD: scrape DOT for agency and contract XLSX tables and create external tables

### LINKED ISSUE #3580: Ingest Safety & Security table endpoints from DOT API and create external tables

### PR DESCRIPTION
**Pull request by charlie-costanzo**
Ingest NTD agency and contractual table endpoints from DOT XLSX files using the pattern developed for the NTD ingest in #3415 

First step in unblocking agency and contractual questions in #3562. Next step is staging and mart tables in the warehouse and metabase

New endpoints:
* 2023_annual_database_agency_information
* 2023_annual_database_contractual_relationship

New tables:
* external_ntd__annual_reporting.2023__annual_database_agency_information
* external_ntd__annual_reporting.2023__annual_database_contractual_relationship

This PR also:
* extends the script to scrape DOT NTD XLSX urls added in #3545 to these new table landing pages, rename script to be table agnostic
  * Although these are annual tables, and the 2023 tables have not yet been updated, tables for previous years were updated intermittently throughout the year and we anticipate this being a possibility for this year's tables as well. In the event that happens, the URL will change, and this script will dynamically pick up the changes
* add test bucket environment variable to docker-compose.yaml for for local development of state geoportal scrape (`CALITP_BUCKET__STATE_GEOPORTAL_DATA_PRODUCTS: "gs://test-calitp-state-geoportal-scrape"`)
* pin `mdformat-gfm==0.3.5` to resolve gh action linter dependency conflicts

Resolves #3580

## Type of change

- [x] New feature

## How has this been tested?
locally using airflow
Screenshot 2024-12-13 at 3 32 22 PM
Screenshot 2024-12-13 at 12 17 41 PM
Screenshot 2024-12-13 at 12 17 36 PM

## Post-merge follow-ups
Monitor for expected behavior, create staging and mart tables
- [x] Actions required (specified below)

### CODE REVIEW

File: airflow/plugins/operators/scrape_ntd_xlsx.py [hunk 1]
```diff
@@ -120,7 +121,15 @@ def execute(self, context, *args, **kwargs):
...
+        if self.product == "annual_database_agency_information":
+            download_url = pull_url_from_xcom(key="agency_url", context=context)
+
+        if self.product == "annual_database_contractual_relationship":
+            download_url = pull_url_from_xcom(
+                key="contractual_relationship_url", context=context
+            )
 
         # see what is returned
         logging.info(f"reading ridership url as {download_url}")
```
Comment (erikamov) [inline airflow/plugins/operators/scrape_ntd_xlsx.py:135]: Maybe needs to be replaced by the product name instead of "ridership".

**Review by erikamov — APPROVED** [review]
Nice work.

File: airflow/plugins/operators/scrape_ntd_xlsx.py (hunk 1 above)
Comment (charlie-costanzo) [reply airflow/plugins/operators/scrape_ntd_xlsx.py]: great catch! just revised

**Comment by vevetron** [comment]
Just wondering what the value of the commented out code is @charlie-costanzo  - should we remove it?

**Comment by vevetron** [comment]
That extra code gets wiped out in a different PR.

**Pull request merged**